### ISSUE TITLE
Unprobed and offline nodes on shortlist

### ISSUE THREAD

**Issue by littleskunk** [Storj/core#618]
### Package Versions

Replace the values below using the output from `npm list storj`. Use ```npm list -g storj``` if installed globally.

```
└─┬ storjshare-cli@9.0.8
  └── storj-lib@6.0.11
```

Replace the values below using the output from `node --version`.

```
v6.9.4
```

### Expected Behavior

Please describe the program's expected behavior. Include an example of your
usage code in the back ticks below if applicable.

I have no idea about the expected behavior.

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

FIND_NODE returns offline nodes and later I try to send messages to these offline nodes. I will remove them from my shortlist. The problem is now that every new FIND_NODE will add them again and again. These 3 offline nodes are closer than anybody else. I have problems with my bloom filter and I can't relay PUBLISH messages.
```
{"level":"debug","message":"searching for nodes close to key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.360Z"}
{"level":"debug","message":"performing network walk for NODE b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.361Z"}
{"level":"debug","message":"starting contact iteration for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.361Z"}
{"level":"debug","message":"querying b37b04d8346d77f76fe2a8f7fde49bfdc23d6845 for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.362Z"}
{"level":"info","message":"sending FIND_NODE message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"010s.storjlt.xyz\",\"port\":6301,\"nodeID\":\"b37b04d8346d77f76fe2a8f7fde49bfdc23d6845\",\"lastSeen\":1483842683362}","timestamp":"2017-01-08T02:31:23.362Z"}
{"level":"debug","message":"queuing callback for reponse to 0dda73b6bd2c378f907494d27cc26d3a47483ab7","timestamp":"2017-01-08T02:31:23.400Z"}
{"level":"debug","message":"querying b36b88ae0a64e119b020607f677742d3759028ef for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.407Z"}
{"level":"info","message":"sending FIND_NODE message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"hdKey\":\"xpub6AHweYHAxk1EhJSBctQD1nLWPog6Sy2eTpKQLExR1hfzTyyZQWvU4EYNXv1NJN7GpLYXnDLt4PzN874g6zSjAQdFCHZN7U7nbYKYVDUzD42\",\"hdIndex\":7443728,\"address\":\"104.196.159.71\",\"port\":8450,\"nodeID\":\"b36b88ae0a64e119b020607f677742d3759028ef\",\"lastSeen\":1483842683407}","timestamp":"2017-01-08T02:31:23.407Z"}
{"level":"debug","message":"queuing callback for reponse to 05022034158ff10f4b4b52e050bb96ccd26d5cbe","timestamp":"2017-01-08T02:31:23.445Z"}
{"level":"debug","message":"querying b34338551e4231e9d6cf16f7bf498f7ce081d2fc for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.451Z"}
{"level":"info","message":"sending FIND_NODE message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client019.storj.dk\",\"port\":15015,\"nodeID\":\"b34338551e4231e9d6cf16f7bf498f7ce081d2fc\",\"lastSeen\":1483842683451}","timestamp":"2017-01-08T02:31:23.451Z"}
{"level":"debug","message":"queuing callback for reponse to e0f014b250149725e9f6e3477e4e6d49ef5a5302","timestamp":"2017-01-08T02:31:23.518Z"}
{"level":"info","message":"received valid message from {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"010s.storjlt.xyz\",\"port\":6301,\"nodeID\":\"b37b04d8346d77f76fe2a8f7fde49bfdc23d6845\",\"lastSeen\":1483842683674}","timestamp":"2017-01-08T02:31:23.674Z"}
{"level":"debug","message":"updating contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"010s.storjlt.xyz\",\"port\":6301,\"nodeID\":\"b37b04d8346d77f76fe2a8f7fde49bfdc23d6845\",\"lastSeen\":1483842683674}","timestamp":"2017-01-08T02:31:23.674Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2017-01-08T02:31:23.674Z"}
{"level":"debug","message":"checking pending rpc callback stack for 0dda73b6bd2c378f907494d27cc26d3a47483ab7","timestamp":"2017-01-08T02:31:23.676Z"}
{"level":"debug","message":"updating contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"010s.storjlt.xyz\",\"port\":6301,\"nodeID\":\"b37b04d8346d77f76fe2a8f7fde49bfdc23d6845\",\"lastSeen\":1483842683361}","timestamp":"2017-01-08T02:31:23.676Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2017-01-08T02:31:23.676Z"}
{"level":"info","message":"received valid message from {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client019.storj.dk\",\"port\":15015,\"nodeID\":\"b34338551e4231e9d6cf16f7bf498f7ce081d2fc\",\"lastSeen\":1483842683688}","timestamp":"2017-01-08T02:31:23.688Z"}
{"level":"debug","message":"updating contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client019.storj.dk\",\"port\":15015,\"nodeID\":\"b34338551e4231e9d6cf16f7bf498f7ce081d2fc\",\"lastSeen\":1483842683688}","timestamp":"2017-01-08T02:31:23.688Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2017-01-08T02:31:23.688Z"}
{"level":"debug","message":"checking pending rpc callback stack for e0f014b250149725e9f6e3477e4e6d49ef5a5302","timestamp":"2017-01-08T02:31:23.689Z"}
{"level":"debug","message":"updating contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client019.storj.dk\",\"port\":15015,\"nodeID\":\"b34338551e4231e9d6cf16f7bf498f7ce081d2fc\",\"lastSeen\":1483842683451}","timestamp":"2017-01-08T02:31:23.690Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2017-01-08T02:31:23.690Z"}
{"level":"info","message":"received valid message from {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"hdKey\":\"xpub6AHweYHAxk1EhJSBctQD1nLWPog6Sy2eTpKQLExR1hfzTyyZQWvU4EYNXv1NJN7GpLYXnDLt4PzN874g6zSjAQdFCHZN7U7nbYKYVDUzD42\",\"hdIndex\":7443728,\"address\":\"104.196.159.71\",\"port\":8450,\"nodeID\":\"b36b88ae0a64e119b020607f677742d3759028ef\",\"lastSeen\":1483842683793}","timestamp":"2017-01-08T02:31:23.793Z"}
{"level":"debug","message":"updating contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"hdKey\":\"xpub6AHweYHAxk1EhJSBctQD1nLWPog6Sy2eTpKQLExR1hfzTyyZQWvU4EYNXv1NJN7GpLYXnDLt4PzN874g6zSjAQdFCHZN7U7nbYKYVDUzD42\",\"hdIndex\":7443728,\"address\":\"104.196.159.71\",\"port\":8450,\"nodeID\":\"b36b88ae0a64e119b020607f677742d3759028ef\",\"lastSeen\":1483842683793}","timestamp":"2017-01-08T02:31:23.793Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2017-01-08T02:31:23.794Z"}
{"level":"debug","message":"checking pending rpc callback stack for 05022034158ff10f4b4b52e050bb96ccd26d5cbe","timestamp":"2017-01-08T02:31:23.795Z"}
{"level":"debug","message":"updating contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"hdKey\":\"xpub6AHweYHAxk1EhJSBctQD1nLWPog6Sy2eTpKQLExR1hfzTyyZQWvU4EYNXv1NJN7GpLYXnDLt4PzN874g6zSjAQdFCHZN7U7nbYKYVDUzD42\",\"hdIndex\":7443728,\"address\":\"104.196.159.71\",\"port\":8450,\"nodeID\":\"b36b88ae0a64e119b020607f677742d3759028ef\",\"lastSeen\":1483842683407}","timestamp":"2017-01-08T02:31:23.795Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2017-01-08T02:31:23.795Z"}
{"level":"debug","message":"finished iteration, handling results","timestamp":"2017-01-08T02:31:23.795Z"}
{"level":"debug","message":"continuing with iterative query for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.795Z"}
{"level":"debug","message":"starting contact iteration for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.796Z"}
{"level":"debug","message":"querying b33082408506539e810a6d460d720be44aa88513 for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.796Z"}
{"level":"info","message":"sending FIND_NODE message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"91.66.32.124\",\"port\":39445,\"nodeID\":\"b33082408506539e810a6d460d720be44aa88513\",\"lastSeen\":1483842683796}","timestamp":"2017-01-08T02:31:23.796Z"}
{"level":"debug","message":"queuing callback for reponse to 6f280a1c3185cb403a788dfbd88def21e03bc35e","timestamp":"2017-01-08T02:31:23.863Z"}
{"level":"debug","message":"querying b3379ccc2881329ff4c30c423e4771408eacaf55 for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.869Z"}
{"level":"info","message":"sending FIND_NODE message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client017.storj.dk\",\"port\":15009,\"nodeID\":\"b3379ccc2881329ff4c30c423e4771408eacaf55\",\"lastSeen\":1483842683869}","timestamp":"2017-01-08T02:31:23.869Z"}
{"level":"debug","message":"queuing callback for reponse to 37734bf8dbb965947b6031b584f08490e8069dd7","timestamp":"2017-01-08T02:31:23.905Z"}
{"level":"debug","message":"querying b337249262d9c9b68a9a2edc7dd7c1e0c3e81954 for key b3328365ee07bfb3a2e71ac8819e2d6c5fcf81e9","timestamp":"2017-01-08T02:31:23.912Z"}
{"level":"info","message":"sending FIND_NODE message to {\"userAgent\":\"6.0.10\",\"protocol\":\"1.0.0\",\"address\":\"symbiont.spdns.de\",\"port\":10300,\"nodeID\":\"b337249262d9c9b68a9a2edc7dd7c1e0c3e81954\",\"lastSeen\":1483842683912}","timestamp":"2017-01-08T02:31:23.912Z"}
{"level":"debug","message":"queuing callback for reponse to 15425ca53e64e6d1e093a9144d0a92037bb3f3c8","timestamp":"2017-01-08T02:31:23.979Z"}
{"level":"warn","message":"error returned from remote host: {\"error\":\"Unable to route to tunnel, client is not connected\"}","timestamp":"2017-01-08T02:31:24.088Z"}
{"level":"warn","message":"missing or empty reply from contact","timestamp":"2017-01-08T02:31:24.088Z"}
{"level":"debug","message":"cleaning bad contacts from routing table","timestamp":"2017-01-08T02:31:34.161Z"}
{"level":"debug","message":"dropping 0 bad contacts from router","timestamp":"2017-01-08T02:31:34.171Z"}
{"level":"info","message":"publishing message on topic \"0e01\"","timestamp":"2017-01-08T02:31:36.711Z"}
{"level":"info","message":"sending PUBLISH message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"010s.storjlt.xyz\",\"port\":6301,\"nodeID\":\"b37b04d8346d77f76fe2a8f7fde49bfdc23d6845\",\"lastSeen\":1483842696713}","timestamp":"2017-01-08T02:31:36.713Z"}
{"level":"debug","message":"queuing callback for reponse to 1f57d5fd5ae22cffce3c3217e3246a27e8400fc4","timestamp":"2017-01-08T02:31:36.752Z"}
{"level":"info","message":"sending PUBLISH message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"hdKey\":\"xpub6AHweYHAxk1EhJSBctQD1nLWPog6Sy2eTpKQLExR1hfzTyyZQWvU4EYNXv1NJN7GpLYXnDLt4PzN874g6zSjAQdFCHZN7U7nbYKYVDUzD42\",\"hdIndex\":7443728,\"address\":\"104.196.159.71\",\"port\":8450,\"nodeID\":\"b36b88ae0a64e119b020607f677742d3759028ef\",\"lastSeen\":1483842696759}","timestamp":"2017-01-08T02:31:36.759Z"}
{"level":"debug","message":"queuing callback for reponse to 5eb7016c6b1da8c4bf409abda9411fedf000e77d","timestamp":"2017-01-08T02:31:36.832Z"}
{"level":"info","message":"sending PUBLISH message to {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client019.storj.dk\",\"port\":15015,\"nodeID\":\"b34338551e4231e9d6cf16f7bf498f7ce081d2fc\",\"lastSeen\":1483842696839}","timestamp":"2017-01-08T02:31:36.839Z"}
{"level":"debug","message":"queuing callback for reponse to 61cad46e473ad2012c9eace3f5837db471293b59","timestamp":"2017-01-08T02:31:36.906Z"}
{"level":"info","message":"received valid message from {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"010s.storjlt.xyz\",\"port\":6301,\"nodeID\":\"b37b04d8346d77f76fe2a8f7fde49bfdc23d6845\",\"lastSeen\":1483842697124}","timestamp":"2017-01-08T02:31:37.124Z"}
{"level":"debug","message":"updating contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"010s.storjlt.xyz\",\"port\":6301,\"nodeID\":\"b37b04d8346d77f76fe2a8f7fde49bfdc23d6845\",\"lastSeen\":1483842697124}","timestamp":"2017-01-08T02:31:37.125Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2017-01-08T02:31:37.125Z"}
{"level":"debug","message":"checking pending rpc callback stack for 1f57d5fd5ae22cffce3c3217e3246a27e8400fc4","timestamp":"2017-01-08T02:31:37.126Z"}
{"level":"debug","message":"checking pending rpc callbacks for expirations","timestamp":"2017-01-08T02:31:37.881Z"}
{"level":"debug","message":"checking pending rpc callbacks for expirations","timestamp":"2017-01-08T02:31:52.887Z"}
{"level":"warn","message":"rpc call 6f280a1c3185cb403a788dfbd88def21e03bc35e timed out","timestamp":"2017-01-08T02:31:52.888Z"}
{"level":"debug","message":"removing contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"91.66.32.124\",\"port\":39445,\"nodeID\":\"b33082408506539e810a6d460d720be44aa88513\",\"lastSeen\":1483842683796}","timestamp":"2017-01-08T02:31:52.888Z"}
{"level":"warn","message":"query failed, removing contact for shortlist, reason RPC with ID `6f280a1c3185cb403a788dfbd88def21e03bc35e` timed out","timestamp":"2017-01-08T02:31:52.888Z"}
{"level":"debug","message":"removing contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"91.66.32.124\",\"port\":39445,\"nodeID\":\"b33082408506539e810a6d460d720be44aa88513\",\"lastSeen\":1483842683796}","timestamp":"2017-01-08T02:31:52.888Z"}
{"level":"warn","message":"rpc call 37734bf8dbb965947b6031b584f08490e8069dd7 timed out","timestamp":"2017-01-08T02:31:52.888Z"}
{"level":"debug","message":"removing contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client017.storj.dk\",\"port\":15009,\"nodeID\":\"b3379ccc2881329ff4c30c423e4771408eacaf55\",\"lastSeen\":1483842683869}","timestamp":"2017-01-08T02:31:52.888Z"}
{"level":"warn","message":"query failed, removing contact for shortlist, reason RPC with ID `37734bf8dbb965947b6031b584f08490e8069dd7` timed out","timestamp":"2017-01-08T02:31:52.889Z"}
{"level":"debug","message":"removing contact {\"userAgent\":\"6.0.11\",\"protocol\":\"1.0.0\",\"address\":\"client017.storj.dk\",\"port\":15009,\"nodeID\":\"b3379ccc2881329ff4c30c423e4771408eacaf55\",\"lastSeen\":1483842683869}","timestamp":"2017-01-08T02:31:52.889Z"}
{"level":"warn","message":"rpc call 15425ca53e64e6d1e093a9144d0a92037bb3f3c8 timed out","timestamp":"2017-01-08T02:31:52.889Z"}
{"level":"debug","message":"removing contact {\"userAgent\":\"6.0.10\",\"protocol\":\"1.0.0\",\"address\":\"symbiont.spdns.de\",\"port\":10300,\"nodeID\":\"b337249262d9c9b68a9a2edc7dd7c1e0c3e81954\",\"lastSeen\":1483842683912}","timestamp":"2017-01-08T02:31:52.889Z"}
{"level":"warn","message":"query failed, removing contact for shortlist, reason RPC with ID `15425ca53e64e6d1e093a9144d0a92037bb3f3c8` timed out","timestamp":"2017-01-08T02:31:52.889Z"}
{"level":"debug","message":"removing contact {\"userAgent\":\"6.0.10\",\"protocol\":\"1.0.0\",\"address\":\"symbiont.spdns.de\",\"port\":10300,\"nodeID\":\"b337249262d9c9b68a9a2edc7dd7c1e0c3e81954\",\"lastSeen\":1483842683911}","timestamp":"2017-01-08T02:31:52.889Z"}
{"level":"debug","message":"finished iteration, handling results","timestamp":"2017-01-08T02:31:52.889Z"}
```

### Steps to Reproduce

Please include the steps the reproduce the issue, numbered below. Include as
much detail as possible.

1. Grep logfile for `removing contact` and watch for the nodeID. I would expect different nodeIDs but at the moment always the same 3 nodeIDs.

**Comment by littleskunk** [comment]
It could be solved with https://github.com/kadtools/kad/issues/89 but that will only reduce the number of FIND_NODE messages. I will be able to relay messages but I will send all 3 SUBSCRIBE message to 3 offline nodes. My bloom filter would be empty and I should not get any contracts.

**Comment by bookchin** [comment]
This should be resolved with #680
